managed/CounterStrikeSharp.API/Core/CoreConfig.cs

```diff
@@ -78,49 +82,58 @@ public static partial class CoreConfig
         public static bool FollowCS2ServerGuidelines => _coreConfig.FollowCS2ServerGuidelines;
     }
 
-    public static partial class CoreConfig
```

Comment:
are you sure that `CoreConfig` should be retrieved as a service instead of `static` access?

Reply:
I am happy to leave it as a service for now, since this is only internal for now and we haven't used it anywhere statically. Ideally anything that reads coreconfig will have access to this through DI

Reply:
we haven't directly (other than the schema field limits) but plugins could access it using `CoreConfig.<option>` without the requirement of DI and services

Reply:
Yeah and that should still work since it has static members with are initialised with the Load method called by core; ideally this class has the same treatment as the game data provider, where we have a static version (like `CoreConfig` is now) but also an instance version which can be injected